src/neuroconv/tools/audio/audio.py

```diff
@@ -0,0 +1,85 @@
+from typing import Optional
+from warnings import warn
+
+from hdmf.backends.hdf5 import H5DataIO
+from ndx_sound import AcousticWaveformSeries
+from pynwb import NWBFile
+
+from neuroconv.tools.hdmf import SliceableDataChunkIterator
+from neuroconv.utils import ArrayType
+
+
+def add_acoustic_waveform_series(
+    acoustic_series: ArrayType,
```

Comment:
Kind of related to question below about automatically fetching info from audio file - what would you think about two tool functions: 

(a) one that takes an `ArrayType` (as you do here) but it's the users burden to figure out how to read that array from some file (and I'd just refer to it generally as `data` or `audio_data` or `audio_array` then)

(b) another that takes as input a `audio_file_path` and handles the data+rate/timestamps extraction for them?

Also, can you remind me what the dimensions are for an audio array? First axis obviously time, but can second dimension be # of audio channels? Or should multiple audio sources be added as multiple `AcousticSeries`?

Reply:
I'd totally agree to separate them. First axis is time, and the second is the number of audio channels, but I haven't seen one with more than one channels. The reason why I thought the Audio interface should take a list of file paths, is assuming what if there is an experiment where different audio data were recorded, and I assumed it could be added as separate `AcousticWaveformSeries`. I think the `AcousticWaveformSeries` itself can hold the audio data with multiple channels. So I was thinking more like multiple recordings, but if that doesn't make sense then the interface should be for a single `file_path`.

Reply:
> but I haven't seen one with more than one channels

@h-mayorquin had an example of more than one channel, however, the underlying storage was as a `.mat` file lol

@h-mayorquin, was that audio data the only objects in the `.mat` file? If so we could easily truncate and add it to the dataset files just as an example of a non-audio format


> The reason why I thought the Audio interface should take a list of file paths, is assuming what if there is an experiment where different audio data were recorded, and I assumed it could be added as separate AcousticWaveformSeries. I think the AcousticWaveformSeries itself can hold the audio data with multiple channels. So I was thinking more like multiple recordings, but if that doesn't make sense then the interface should be for a single file_path.

This is where there's a question, similar to the `VideoInterface`. I'd be fine with the interface itself being a list of `file_paths` (just like the `VideoInterface`) but the thing to pay attention to is without an API like SI/ROIEx it's hard to specify if the files are separated over time or channels. I think what we did in the `VideoInterface` was the capability of a list of list of file paths, with each inner list being concatenated in time and the outer one being other `ImageSeries`. That's more of an advanced case here though, I'd say let's keep it simple for now.